ZenPacks/zenoss/ZenPackLib/lib/base/ComponentBase.py

```diff
@@ -49,6 +49,32 @@ class ComponentBase(ModelBase):
             }
         }
 
+    def __setattr__(self, name, value):
+        '''enforce type checking when setting _properties attributes'''
+        # dictionary of _property "type" keys and target Python classes
+        target_type_map = {'boolean': bool,
+                           'int': int,
+                           'float': float,
+                           'string': str,
+                           'password': str,
+                           }
+        # dictionary of _properties "id"s and "type"s
+        property_dict = dict([(p.get('id'), p.get('type')) for p in self._properties])
+        # only change type if it's a _properties attribute
+        if name in property_dict.keys():
```

Comment:
This nested checking of dictionary contents could be replaced with the following.

```python
target_class = target_type_map.get(property_dict.get(name))
if target_class:
    try:
        value = target_class(value)
    # etc..
```

In cases where a dict value can't be None it's usually better to "get" before "if" instead of "if in" before "get".

Also, there's no need to use the _keys_ method of a dict when doing an "in" check. It creates an unnecessary temporary list of the dictionaries keys when you can just do "if key in mydict" directly.

Lastly on the "except Exception", don't add the "as e" part if you aren't going to use e since it becomes an unused variable.